app/gameengine.py

```diff
@@ -1903,6 +1904,16 @@ def get_available_mainstep_actions(self):
                     if not self.are_conditions_met(active_player, card["game_card_id"], card["play_conditions"]):
                         continue
 
+                # Baked in restriction for mascots and tools
+                if is_card_mascot(card) or is_card_tool(card):
+                    sub_type = card["sub_type"]
+                    # Checks if there are available holomem on stage that can still be attached with the support card
+                    if not self.is_condition_met(active_player, card["game_card_id"],{
+                        "condition": "holomem_on_stage_can_be_attached_with",
```

Comment:
I think I would generally prefer you to just make a function that checks the condition rather than making a new condition field and doing it this way.
It's just an extra step we don't need.

If you want to leave the condition because you're adding something in another change, just make it call the same function?